### PR TITLE
Add CODEOWNERS file

### PR DESCRIPTION
**Pull request by raulchen**
## What do these changes do?

Add the `CODEWONERS` file, which defines the owners of each module. When a PR touches a file of a module, the corresponding owners will be automatically added as reviewers. 

## Linter

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by raulchen** [comment]
The owners are grouped by teams, e.g. `@ray-project/ray-core-cpp`. I haven't created those teams yet. Please let me know if you have better ideas about how to categorize the code.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+# ==== Build and CI ====
+
+# Bazel.
+/BUILD.bazel @ray-project/ray-core
+/WORKSPACE @ray-project/ray-core
+/bazel/ @ray-project/ray-core
+
+# CI.
+/ci/ @ray-project/ray-core
```
Comment (ericl) [inline .github/CODEOWNERS]: I would exclude ci/ since a lot of RLlib / tune / misc tests go through this.

Comment (raulchen) [reply .github/CODEOWNERS]: Changed this to only include the CI scripts.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# Python worker.
+/python/ray/ @ray-project/ray-core-python
+!/python/ray/tune/ @ray-project/ray-core-python
+!/python/ray/rllib/ @ray-project/ray-core-python
+!/python/ray/modin/ @ray-project/ray-core-python
+
+# Java worker.
+/java/ @ray-project/ray-core-java
+
+# ==== Llibraries and frameworks ====
```
Comment (robertnishihara) [inline .github/CODEOWNERS]:
```suggestion
# ==== Libraries and frameworks ====
```

**Comment by robertnishihara** [comment]
It might make sense to make it even more fine-grained, e.g., also include the autoscaler and the build system as two separate groups. We can also try what you have and see if it's helping and iterate.

**Comment by raulchen** [comment]
yeah, we can try this first and make it more fine-grained later.
Comments are addressed.

**Comment by raulchen** [comment]
@robertnishihara @ericl any other comments?

**Review by ericl — APPROVED** [review]
Looks good, let's try it out

**Pull request merged**